GVFS/GVFS.Notifications/VFSForGit.Mac/StatusMenuItem/UI/VFSAboutWindowController.m

```diff
@@ -0,0 +1,55 @@
+#import "VFSAboutWindowController.h"
+
+@interface VFSAboutWindowController ()
+
+@property (strong) VFSProductInfoFetcher *productInfoFetcher;
+
+@end
+
+@implementation VFSAboutWindowController
+
+- (instancetype)initWithProductInfoFetcher:(VFSProductInfoFetcher *)productInfoFetcher
+{
+    if (productInfoFetcher == nil)
+    {
+        self = nil;
```

Comment:
When will we hit this code path? Is this an error condition? (mainly I am not familiar with this pattern in obj-c...)

Reply:
The pattern is to return a self=nil when required conditions are not met during object initialization. 
Foo *foo = [[Foo alloc] init];
A Foo class sized memory location is allocated in alloc, and if init fails then it needs to return a nil instead of an incompletely initialized object. Setting self=nil clears that memory (in ARC), and prevents super class' init from being called. 
https://developer.apple.com/documentation/objectivec/nsobject/1418641-init?language=objc
https://stackoverflow.com/questions/4336953/returning-nil-in-the-init-in-objective-c-code